yo1k/tic_tac_toe/test/core/test_game.py

```diff
@@ -10,90 +10,88 @@ def __init__(self, actions: Sequence[Optional[g.Action]]):
         self.i = 0
 
     def next(self) -> Optional[g.Action]:
-        if self.i > len(self.actions):
+        if self.i >= len(self.actions):
             return None
         action = self.actions[self.i]
         self.i += 1
         return action
 
 
-class StateSetup:
-    @staticmethod
-    def _set_state(
-            game_rounds=5,
-            player1=None,
-            player2=None,
-            board=None,
-            phase=g.Phase.INROUND,
-            cur_round=0,
-            step=0) -> g.State:
-        player1 = g.Player(g.Mark.X) if player1 is None else player1
-        player2 = g.Player(g.Mark.O) if player2 is None else player2
-        board = g.Board() if board is None else board
-        return g.State(game_rounds, [player1, player2], board, phase, cur_round, step)
+def _new_state(
+        game_rounds=5,
+        player1=None,
+        player2=None,
+        board=None,
+        phase=g.Phase.INROUND,
+        cur_round=0,
+        step=0) -> g.State:
```

Comment:
I thought you said that there is no double indentation for parts of a code string when it is split into multiple lines in Python. Is this recollection incorrect?

Reply:
I was wrong. This recollection is correct (see also [example](https://www.python.org/dev/peps/pep-0008/#:~:text=def%20long_function_name(%0A%20%20%20%20%20%20%20%20var_one%2C%20var_two%2C%20var_three%2C%0A%20%20%20%20%20%20%20%20var_four)%3A%0A%20%20%20%20print(var_one)) in PEP8)